src/main/java/racingcar/game/Game.java

```diff
@@ -0,0 +1,89 @@
+package racingcar.game;
+
+import racingcar.io.InputAdaptor;
+import racingcar.io.Printer;
+import racingcar.util.OutputFormatter;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a game in run. Game class is responsible for receiving
+ * user inputs, executing a game, and output in appropriate situation.
+ * First you should call two methods to invoke the user input needed,
+ * and then call execute() method to run a game.
+ */
+public class Game {
+    private static final String UNINITIALIZED_EXCEPTION_MSG = "사용자 입력이 이루어지지 않았습니다.";
+
+    private final InputAdaptor input;
+
+    private GameManager gameManager;
+    private String[] cars;
+    private int numOfTries;
+
+    /**
+     * Constructor.
+     * @param input an InputAdaptor object
+     */
+    public Game(InputAdaptor input) {
+        this.input = input;
+        this.gameManager = null;
+        this.numOfTries = -1;
+    }
+
+    /**
+     * Let the user enter names of cars.
+     * The user input has to be the following form:
+     * car1,car2,car3,...
+     * Each car name is separated by comma(,).
+     * A message saying that the user has to input and how the user input is printed.
+     */
+    public void inputCarNames() {
+        Printer.print(OutputFormatter.getMessageWhenInputCarNames());
+        String[] carNames = input.readCarNames();
+        this.cars = carNames;
+        this.gameManager = new GameManager(carNames);
+    }
+
+    /**
+     * Let the user enter the number what times steps of the game are executed.
+     * A message saying that the user has to input and how the user input is printed.
+     */
+    public void inputNumOfTries() {
+        Printer.print(OutputFormatter.getMessageWhenInputNumOfTries());
+        this.numOfTries = input.readNumOfTries();
+    }
+
+    /**
+     * Executes a game. If the user hasn't entered car names and
+     * how many tries would run, an IllegalStateException will be thrown.
+     */
+    public void execute() {
+        if (this.numOfTries < 0 || this.gameManager == null) {
+            throw new IllegalStateException(UNINITIALIZED_EXCEPTION_MSG);
+        }
```

Comment:
지금도 좋지만, 제가 보기엔 분기의 각 조건을 메서드로 분리하면 의미 전달이 더 잘 될 것 같습니다 !

```suggestion
    public void execute() {
        if (hasInvalidNumberOfTries() || isCarNamesMissing()) {
            throw new IllegalStateException(UNINITIALIZED_EXCEPTION_MSG);
        }
```

Reply:
미처 생각지 못한 부분이네요. 확실히 제시해 주신 대로 하는 편이 더 의미 전달이 잘 되는 것 같아요. 좋은 의견 감사합니다!